ui/app/models/config-ui/message.js

```diff
@@ -2,21 +2,96 @@
  * Copyright (c) HashiCorp, Inc.
  * SPDX-License-Identifier: BUSL-1.1
  */
-
 import Model, { attr } from '@ember-data/model';
 import lazyCapabilities, { apiPath } from 'vault/macros/lazy-capabilities';
-import { isAfter } from 'date-fns';
+import { isAfter, format, addDays, startOfDay } from 'date-fns';
 import { parseAPITimestamp } from 'core/utils/date-formatters';
+import { withModelValidations } from 'vault/decorators/model-validations';
+import { withFormFields } from 'vault/decorators/model-form-fields';
+
+export const localDateTimeString = "yyyy-MM-dd'T'HH:mm";
+
+const validations = {
+  title: [{ type: 'presence', message: 'Title is required.' }],
+  message: [{ type: 'presence', message: 'Message is required.' }],
+};
 
+@withFormFields(['authenticated', 'type', 'title', 'message', 'linkTitle', 'startTime', 'endTime'])
+@withModelValidations(validations)
 export default class MessageModel extends Model {
   @attr('boolean') active;
-  @attr('string') type;
-  @attr('boolean') authenticated;
-  @attr('string') title;
-  @attr('string') message;
-  @attr('object') link;
-  @attr('string') startTime;
-  @attr('string') endTime;
+  @attr('string', {
+    label: 'Type',
+    editType: 'radio',
+    possibleValues: [
+      {
+        label: 'Alert message',
+        subText:
+          'A banner that appears on the top of every page to display brief but high-signal messages like an update or system alert.',
+        value: 'banner',
+      },
+      {
+        label: 'Modal',
+        subText: 'A pop-up window used to bring immediate attention for important notifications or actions.',
+        value: 'modal',
+      },
+    ],
+    defaultValue: 'banner',
+  })
+  type;
+  // The authenticated attr is a boolean. The authenticatedString getter and setter is used to get the and set
+  // the values as strings. The server and query params expects the attr to be boolean values.
```

Comment:
Thank you for adding a comment! Small typo (an extra `the`), also made a wording suggestion if you want! (Since it kind of sounds like "setting as strings" means we're setting the model attr as a string  
```suggestion
  // The authenticated attr is a boolean. The authenticatedString getter and setter is used only in forms to get and set the boolean via strings values. The server and query params expects the attr to be boolean values.
```